### PR TITLE
Add `regexp/no-standalone-backslash` rule

### LINKED ISSUE #129: New rule to forbid literal backslashes
In non-Unicode mode, the regex `/\c/` is equivalent to `/\\c/`. The `\` in `/\c/` is a literal backslash. It would probably be good to have a rule that detects this rather strange behavior.

### PR DESCRIPTION
**Pull request by ota-meshi**
close #129

### CODE REVIEW

**Comment by ota-meshi** [comment]
There seems to be a problem with GitHub Actions right now. I will re-run it later.
https://www.githubstatus.com/incidents/cj7gzzj30411

File: lib/rules/no-standalone-reverse-solidus.ts [hunk 1]
```diff
@@ -0,0 +1,45 @@
+import type { RegExpVisitor } from "regexpp/visitor"
+import type { RegExpContext } from "../utils"
+import { createRule, defineRegexpVisitor, CP_BACK_SLASH } from "../utils"
+
+export default createRule("no-standalone-reverse-solidus", {
+    meta: {
+        docs: {
+            description: "disallow standalone reverse solidus (`\\`)",
```
Comment (RunDevelopment) [inline lib/rules/no-standalone-reverse-solidus.ts:8]: This rule should probably be renamed. I had no idea what a "reverse-solidus" was until now. "Backslash" is more commonly used.

File: docs/rules/no-standalone-reverse-solidus.md [hunk 2]
```diff
@@ -0,0 +1,55 @@
...
+- :exclamation: <badge text="This rule has not been released yet." vertical="middle" type="error"> ***This rule has not been released yet.*** </badge>
+
+## :book: Rule Details
+
+This rule disallows reverse solidus (`\`) without escape.
+
+The regular expression `/\c/` without the unicode (`u`) flag has the same pattern as `/\\c/`.
+
+This is the specification described in [Annex B] of ECMAScript. But very confusing and you should not use it intentionally.
+In most cases you intended the control escape sequence (`\cX`) or another escape, but you maybe made a mistake.
```
Comment (RunDevelopment) [inline docs/rules/no-standalone-reverse-solidus.md:20]: I change the text a little.

```suggestion
E.g. the regular expression `/\c/` without the unicode (`u`) flag is the same pattern as `/\\c/`.

In most cases, standalone backslashes are used by accident when a control escape sequence (`\cX`) or another escape sequence was intended. They are very confusing and should not be used intentionally.

This behavior is described in [Annex B] of the ECMAScript specification.
```

**Review by RunDevelopment — CHANGES_REQUESTED** [review]

File: lib/rules/no-standalone-reverse-solidus.ts (hunk 1 above)
Comment (ota-meshi) [reply lib/rules/no-standalone-reverse-solidus.ts:8]: I personally liked `backslash`, but the name in the spec was `reverse-solidus`. I didn't know which one was easier for people to understand, so I followed the spec.
But you and I think backslash is good so change to `backslash`!

File: docs/rules/no-standalone-reverse-solidus.md (hunk 2 above)
Comment (ota-meshi) [reply docs/rules/no-standalone-reverse-solidus.md:20]: Thank you!

**Comment by ota-meshi** [comment]
Thank you for your review! I renamed the rule to `regexp/no-standalone-backslash`.

File: lib/rules/no-standalone-backslash.ts [hunk 3]
```diff
@@ -0,0 +1,45 @@
+import type { RegExpVisitor } from "regexpp/visitor"
+import type { RegExpContext } from "../utils"
+import { createRule, defineRegexpVisitor, CP_BACK_SLASH } from "../utils"
+
+export default createRule("no-standalone-backslash", {
+    meta: {
+        docs: {
+            description: "disallow standalone backslash (`\\`)",
```
Comment (RunDevelopment) [inline lib/rules/no-standalone-backslash.ts:8]: It has to be plural here.

```suggestion
            description: "disallow standalone backslashes (`\\`)",
```

**Review by RunDevelopment — CHANGES_REQUESTED** [review]
One last thing and then it's good to go.

File: lib/rules/no-standalone-backslash.ts (hunk 3 above)
Comment (ota-meshi) [reply lib/rules/no-standalone-backslash.ts]: Thank you!

**Review by RunDevelopment — APPROVED** [review]

**Pull request merged**